### PR TITLE
Punctuation capitalization model docs

### PR DESCRIPTION
**Pull request by jubick1337**
# What does this PR do ?

Adds docs for p&c lexical audio model

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [X] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst [hunk 1]
```diff
@@ -0,0 +1,953 @@
...
+
+.. code-block:: python
+
+    from nemo.collections.nlp.models import PunctuationCapitalizationLexicalAudioModel
+
+    # to get the list of pre-trained models
+    PunctuationCapitalizationLexicalAudioModel.list_available_models()
+
+    # Download and load the pre-trained BERT-based model
+    model = PunctuationCapitalizationLexicalAudioModel.from_pretrained("PLACEHOLDER")
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:29]: use `restore_from .. <PATH to .nemo file>` until chpts are released

**Review by ekmb — CHANGES_REQUESTED** [review]
This doc should be added to the NLP tasks list. How about refactoring similar to this doc: https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/nlp/text_normalization/neural_models.html? P&C task page that has a link to the default and default + acoustic models?

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst [hunk 2]
```diff
@@ -0,0 +1,953 @@
...
+    model.add_punctuation_capitalization(['how are you', 'great how about you'], audio_queries=['/path/to/1.wav', '/path/to/2.wav'], target_sr=16000)
+
+Model Description
+-----------------
+In the addition to Punctuation and Capitalization model we add audio encoder (e.g. Conformer's encoder) and attention based fusion of lexical and audio features.
+This model architecture is based on `Multimodal Semi-supervised Learning Framework for Punctuation Prediction in Conversational Speech <https://arxiv.org/pdf/2008.00702.pdf>`_.
+
+.. note::
+
+    We recommend you try this model in a Jupyter notebook (run on `Google's Colab <https://colab.research.google.com/notebooks/intro.ipynb>`_.): `NeMo/tutorials/nlp/Punctuation_and_Capitalization_Lexical_Audio.ipynb <https://github.com/NVIDIA/NeMo/blob/stable/tutorials/nlp/Punctuation_and_Capitalization_Lexical_Audio.ipynb>`__.
```
[the remark this answers is not in the document]
Comment (jubick1337) [reply docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:41]: @ekmb not really but it's more of a common problem, nothing to do with tutorial itself IMO

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+    python punctuate_capitalize_infer.py \
+        --input_manifest <PATH/TO/INPUT/MANIFEST> \
+        --output_manifest <PATH/TO/OUTPUT/MANIFEST> \
+        --pretrained_name PLACEHOLDER \
+        --max_seq_length 64 \
+        --margin 16 \
+        --step 8 \
+        --use_audio
+
+Long audios are split into corresponding sequences of :code:`4000 * max_seq_length`.
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:352]: what does this mean? what are audio/text files restrictions for this model?

File: docs/source/nlp/punctuation_and_capitalization_models.rst [hunk 4]
```diff
@@ -0,0 +1,31 @@
+.. _punctuation_capitalization_models:
+
+Punctuation And Capitalization Models
+==============================================
+
+NeMo provides two types of Punctuation And Capitalization Models:
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization_models.rst:6]: this should be moved after line 15

File: docs/source/nlp/punctuation_and_capitalization.rst [hunk 5]
```diff
@@ -80,6 +72,30 @@ The ``source_data_dir`` structure should look similar to the following:
 
 .. _nemo-data-format-label:
 
+The Punctuation and Capitalization Lexical Audio model also requires audio data corresponding to content of ``train.txt/dev.txt/test.txt``. 
+Audio data should be stored in ``audio_train.txt`` and ``audio_dev.txt`` (and optionally ``test.txt```). Each line in the ``audio_train.txt/audio_dev.txt/audio_test.txt`` should represent one calid path to audio file.
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization.rst:76]: why are these audio-related format details in the text-only model?

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekmb) [reply docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:43]: please re-run, the default tutorial works in colab

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+-----------------
+
+.. code-block:: python
+
+    from nemo.collections.nlp.models import PunctuationCapitalizationLexicalAudioModel
+
+    # to get the list of pre-trained models
+    PunctuationCapitalizationLexicalAudioModel.list_available_models()
+
+    # Download and load the pre-trained BERT-based model
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:30]: irrelevant comment

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+    model = PunctuationCapitalizationLexicalAudioModel.from_pretrained("<PATH to .nemo file>")
+
+    # try the model on a few examples
+    model.add_punctuation_capitalization(['how are you', 'great how about you'], audio_queries=['/path/to/1.wav', '/path/to/2.wav'], target_sr=16000)
+
+Model Description
+-----------------
+In the addition to Punctuation and Capitalization model we add audio encoder (e.g. Conformer's encoder) and attention based fusion of lexical and audio features.
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:38]: add link to the text only model here

**Review by ekmb — CHANGES_REQUESTED** [review]

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+You can freeze audio encoder during training and add additional ``ConformerLayer`` on top of encoder to reduce compute with ``model.audio_encoder.freeze``. You can also add `Adapters <https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/core/adapters/components.html>`_ to reduce compute with ``model.audio_encoder.adapter``. Parameters of fusion module are stored in ``model.audio_encoder.fusion``.
+An example of a model configuration file for training the model can be found at:
+`NeMo/examples/nlp/token_classification/conf/punctuation_capitalization_lexical_audio_config.yaml <https://github.com/NVIDIA/NeMo/blob/stable/examples/nlp/token_classification/conf/punctuation_capitalization_lexical_audio_config.yaml>`__.
+
+Configs
+^^^^^^^^^^^^
+.. note::
+  This page contains only paraemters specific to lexical and audio model. Others parameters can be found in `Punctuation And Capitalization's page <https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/nlp/punctuation_and_capitalization.html>`_.
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:113]: typo in `paraemters`

File: docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+    python punctuate_capitalize_infer.py \
+        --input_manifest <PATH/TO/INPUT/MANIFEST> \
+        --output_manifest <PATH/TO/OUTPUT/MANIFEST> \
+        --pretrained_name PLACEHOLDER \
+        --max_seq_length 64 \
+        --margin 16 \
+        --step 8 \
+        --use_audio
+
+Long audios are split just like in text only case, audio sequences treated the same as text seqences except :code:`max_seq_length` for audio equals :code:`max_seq_length*4000`.
```
Comment (ekmb) [inline docs/source/nlp/punctuation_and_capitalization_lexical_audio.rst:352]: how are these text/audio chucks are aligned?

**Pull request closed without merging**